src/execution/execute-afm.convert.ts

```diff
@@ -0,0 +1,75 @@
+// (C) 2007-2019 GoodData Corporation
+
+import { AFM, ExecuteAFM } from '@gooddata/typings';
+import { unset, cloneDeep } from 'lodash';
+
+function convertElementsArray(arr: string[], isText?: boolean): ExecuteAFM.AttributeElements {
+    return isText ? { values: arr } : { uris: arr };
+}
+
+function convertFilter(filter: AFM.CompatibilityFilter): ExecuteAFM.CompatibilityFilter {
+    if (AFM.isPositiveAttributeFilter(filter)) {
+        const clonedFilter = cloneDeep(filter) as ExecuteAFM.IPositiveAttributeFilter;
+        const { positiveAttributeFilter: oldFilter } = filter;
+        const { positiveAttributeFilter: newFilter } = clonedFilter;
+
+        newFilter.in = convertElementsArray(oldFilter.in, oldFilter.textFilter);
+        unset(newFilter, 'textFilter');
+
+        return { positiveAttributeFilter: newFilter };
+    } else if (AFM.isNegativeAttributeFilter(filter)) {
+        const clonedFilter = cloneDeep(filter) as ExecuteAFM.INegativeAttributeFilter;
+        const { negativeAttributeFilter: oldFilter } = filter;
+        const { negativeAttributeFilter: newFilter } = clonedFilter;
+
+        newFilter.notIn = convertElementsArray(oldFilter.notIn, oldFilter.textFilter);
+        unset(newFilter, 'textFilter');
+
+        return { negativeAttributeFilter: newFilter };
+    }
+
+    return filter;
+}
+
+function convertFilters(filters?: AFM.CompatibilityFilter[]): ExecuteAFM.CompatibilityFilter[] | undefined {
+    return filters !== undefined ? filters.map(convertFilter) : filters;
+}
+
+/**
+ * Converts 'client-land' AFM to one that will be sent to backend.
+ *
+ * @param afm afm to convert
+ */
+export function convertAfm(afm: AFM.IAfm | undefined): ExecuteAFM.IAfm {
+    if (afm === undefined) {
+        return {};
+    }
+
+    const clonedAfm = cloneDeep(afm) as ExecuteAFM.IAfm;
```

Comment:
I have a same concern about this as mentioned above with filters... I would rather to avoid usage of "cloneDeep", "as" and other mutating functions... what do you think? :)